packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -129,10 +129,10 @@ class CustomSemanticsAction {
   bool operator ==(dynamic other) {
     if (other.runtimeType != runtimeType)
       return false;
-    final CustomSemanticsAction typedOther = other;
-    return typedOther.label == label
-      && typedOther.hint == hint
-      && typedOther.action == action;
+    return other is CustomSemanticsAction
```

Comment:
Same as in the other PRs, we already know that other is assignable to CustomSemanticsAction.